packages/knip/src/plugins/hardhat/index.ts

```diff
@@ -1,25 +1,58 @@
-import type { IsPluginEnabled, Plugin, Resolve } from '../../types/config.js';
-import { toDependency } from '../../util/input.js';
-import { hasDependency } from '../../util/plugin.js';
+import type {
+  IsPluginEnabled,
+  Plugin,
+  Resolve,
+  ResolveConfig,
+} from "../../types/config.js";
+import { toDeferResolve, toDependency, toEntry } from "../../util/input.js";
+import { hasDependency } from "../../util/plugin.js";
+import type { HardhatUserConfig } from "./types.js";
 
 // https://hardhat.org/docs
 
-const title = 'Hardhat';
+const title = "Hardhat";
 
-const enablers = ['hardhat'];
+const enablers = ["hardhat"];
 
-const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+const isEnabled: IsPluginEnabled = ({ dependencies }) =>
+  hasDependency(dependencies, enablers);
 
-const entry: string[] = ['hardhat.config.{js,cjs,mjs,ts}'];
+const config = ["hardhat.config.{js,cjs,mjs,ts}"];
 
 const resolve: Resolve = async () => {
-  return [toDependency('hardhat')];
+  const inputs = [toDependency("hardhat")];
+
+  // TODO: only add test files if a Hardhat test reporter is installed
```

Comment:
Is it possible to check for the presence of other packages here, or does that require a separate plugin?